cadence/contracts/bridge/IFlowEVMNFTBridge.cdc

```diff
@@ -0,0 +1,115 @@
+    *********************************/
+
+    /// Public entrypoint to bridge NFTs from Cadence to EVM.
+    ///
+    /// @param token: The NFT to be bridged
+    /// @param to: The NFT recipient in FlowEVM
+    /// @param feeProvider: A reference to a FungibleToken Provider from which the bridging fee is withdrawn in $FLOW
+    ///
+    access(all)
+    fun bridgeNFTToEVM(
+        token: @{NonFungibleToken.NFT},
+        to: EVM.EVMAddress,
+        feeProvider: auth(FungibleToken.Withdraw) &{FungibleToken.Provider}
+    ) {
+        pre {
+            emit BridgedNFTToEVM(
+                type: token.getType(),
+                id: token.id,
+                evmID: CrossVMNFT.getEVMID(from: &token as &{NonFungibleToken.NFT}) ?? UInt256(token.id),
+                to: FlowEVMBridgeUtils.getEVMAddressAsHexString(address: to),
+                evmContractAddress: FlowEVMBridgeUtils.getEVMAddressAsHexString(
+                    address: self.getAssociatedEVMAddress(with: token.getType())
+                        ?? panic("Could not find EVM Contract address associated with provided NFT")
+                ), bridgeAddress: self.account.address
+            )
+        }
+    }
+
+    /// Public entrypoint to bridge NFTs from EVM to Cadence
+    ///
+    /// @param owner: The EVM address of the NFT owner. Current ownership and successful transfer (via 
+    ///     `protectedTransferCall`) is validated before the bridge request is executed.
+    /// @param calldata: Caller-provided approve() call, enabling contract COA to operate on NFT in EVM contract
+    /// @param id: The NFT ID to bridged
+    /// @param evmContractAddress: Address of the EVM address defining the NFT being bridged - also call target
+    /// @param feeProvider: A reference to a FungibleToken Provider from which the bridging fee is withdrawn in $FLOW
+    /// @param protectedTransferCall: A function that executes the transfer of the NFT from the named owner to the
+    ///     bridge's COA. This function is expected to return a Result indicating the status of the transfer call.
+    ///
+    /// @returns The bridged NFT
+    ///
+    access(all)
+    fun bridgeNFTFromEVM(
+        owner: EVM.EVMAddress,
+        type: Type,
+        id: UInt256,
+        feeProvider: auth(FungibleToken.Withdraw) &{FungibleToken.Provider},
+        protectedTransferCall: fun (): EVM.Result
+    ): @{NonFungibleToken.NFT} {
+        post {
+            emit BridgedNFTFromEVM(
+                type: result.getType(),
+                id: result.id,
+                evmID: id,
+                caller: FlowEVMBridgeUtils.getEVMAddressAsHexString(address: owner),
+                evmContractAddress: FlowEVMBridgeUtils.getEVMAddressAsHexString(
+                    address: self.getAssociatedEVMAddress(with: result.getType())
+                        ?? panic("Could not find EVM Contract address associated with provided NFT")
+                ), bridgeAddress: self.account.address
```

Comment:
Should we also have the bridgeContractName field here?

Reply:
> For simplicity purposes of the EVM bridge as a whole though, are we sure we want to allow other bridges to emit these exact same events?

> Is there a use-case inspiring this, or is it just to allow for future similar use-cases? Are there some risks we're introducing from this, or is the guidance that consumers can just filter based on contract address and its all good?

It's a good question. I don't think we'd need to worry about other contracts impersonating the central bridge since listeners could filter on `bridgeAddress` which the interface guarantees will be accurate. Curious if anyone can think of vulnerabilities this might introduce.

Regarding use case - I thought it would be helpful at minimum which prototyping the bridge on Previewnet. In the event we need to deploy updates to a new account, listeners could just filter on the updated address. Ideally, we'd get visibility into all assets bridging between VMs with this interface, but the reversion on `evmContractAddress` likely limits the usefulness of this interface to just this public infra bridge.

> Should we also have the bridgeContractName field here?

I'm not sure there's a way to dynamically retrieve the contract name from within the contract at runtime 🤔 But I could see it being helpful.